servatrice/src/servatrice.cpp

```diff
@@ -515,9 +514,9 @@ QList<AbstractServerSocketInterface *> Servatrice::getUsersWithAddressAsList(con
 {
     QList<AbstractServerSocketInterface *> result;
     QReadLocker locker(&clientsLock);
-    for (int i = 0; i < clients.size(); ++i)
-        if (static_cast<AbstractServerSocketInterface *>(clients[i])->getPeerAddress() == address)
-            result.append(static_cast<AbstractServerSocketInterface *>(clients[i]));
+    for (auto client : clients)
+        if (dynamic_cast<AbstractServerSocketInterface *>(client)->getPeerAddress() == address)
+            result.append(dynamic_cast<AbstractServerSocketInterface *>(client));
```

Comment:
A `client` is a `Server_ProtocolHandler *`, and the`AbstractServerSocketInterface` is a subclass of `Server_ProtocolHandler`.
Since we can guarantee that the down-cast is safe, `static_cast` is a better choice.
`dynamic_cast` could be a safer choice if we are not sure about the type of `client`, but still we should check the cast result before dereferencing it.
Also, `dynamic_cast` has a runtime performance cost.

Reply:
CLion didn't agree with this being static for some reason, but I'll listen